Assets/Scripts/skeletonScript.cs

```diff
@@ -144,18 +145,30 @@ private GameObject DetectClosestEnemy()
                 targetPos = enemiesInRange[i].gameObject;
             }
         }
-        return targetPos;
+        return targetPos.GetComponent<Enemies>();
     }
 
     protected override void Attack()
     {
-        
+        Actions.OnEnemyDamaged?.Invoke(targetEnemy, damage);
```

Comment:
Remove the Action and just call targetEnemy.TakeDamage(damage)

Reply:
Done